src/test/java/com/zalando/ECommerceManagement/userManagement/controller/ProductControllerTest.java

```diff
@@ -171,4 +177,97 @@ public void testDeleteProductById() throws Exception {
 
         verify(productService, times(1)).deleteProduct(1);
     }
+
+    @ParameterizedTest
+    @MethodSource("getProductInput")
+    @DisplayName("Should not create a new product with null fields")
+    public void testInvalidProductFields(
+            Product newProduct,
+            String errorKey,
+            String errorMessage) throws Exception {
+        String jsonRequest = new ObjectMapper().writeValueAsString(newProduct);
+        mockMvc.perform(post("/products").contentType(MediaType.APPLICATION_JSON)
+                        .content(jsonRequest))
+                .andExpect(status().isBadRequest())
+                .andExpect(MockMvcResultMatchers.jsonPath("$." + errorKey).value(errorMessage));
+
+        verify(productService, times(0)).createProduct(any(Product.class));
+    }
+
+    static Stream<Arguments> getProductInput() {
```

Comment:
can you move your non-test functions to the bottom? Also you can make this private since it is used just in this class